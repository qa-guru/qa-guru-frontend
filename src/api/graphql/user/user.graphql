query user {
  user {
    id
    email
    firstName
    lastName
    middleName
    phoneNumber
    roles
    locked
    vkId
    git
    telegram
    stackOverflow
    linkedin
    website
    avatar
    skills
    creationDate
    confirmationDate
    updateDate
    rating {
      rating
    }
  }
}
