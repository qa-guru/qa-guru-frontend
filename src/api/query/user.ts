import {
  UserQuery,
  UserQueryVariables,
  useUserQuery as _useUserQuery,
} from "../../generated/graphql";
import * as Apollo from "@apollo/client";
import { ApolloError } from "@apollo/client/errors";
import { useSnackbar } from "notistack";

export const useUserQuery = (
  baseOptions?: Apollo.QueryHookOptions<UserQuery, UserQueryVariables>
) => {
  const { enqueueSnackbar } = useSnackbar();

  return _useUserQuery({
    onError: (error: ApolloError) =>
      error.graphQLErrors.map(({ message }) => enqueueSnackbar(message)),
    ...baseOptions,
  });
};
