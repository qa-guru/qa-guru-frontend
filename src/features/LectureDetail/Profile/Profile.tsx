import React from "react";
import { Box, Stack, Typography } from "@mui/material";
import Avatar from "@mui/material/Avatar";
import { format, parseISO } from "date-fns";
import { IProfile } from "./Profile.types";

const style = {
  avatar: {
    width: 40,
    height: 40,
  },
};

const Profile: React.FC<IProfile> = (props) => {
  const { firstName, lastName, date } = props;

  return (
    <Stack spacing={1.7} direction="row" alignItems="center">
      <Avatar
        sx={style.avatar}
        alt="Remy Sharp"
        src="/static/images/avatar/1.jpg"
      />
      <Box>
        <Typography variant="subtitle1">
          {firstName} {lastName}
        </Typography>
        <Stack direction="row" justifyContent="space-between">
          <Typography variant="subtitle2">
            {format(parseISO(date), "dd.MM.yyyy '|' HH:mm")}
          </Typography>
        </Stack>
      </Box>
    </Stack>
  );
};

export default Profile;
