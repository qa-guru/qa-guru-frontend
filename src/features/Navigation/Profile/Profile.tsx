import React from "react";
import {
  Box,
  Button,
  ListItemIcon,
  ListItemText,
  MenuItem,
  MenuList,
  Typography,
} from "@mui/material";
import Tooltip from "@mui/material/Tooltip";
import Avatar from "@mui/material/Avatar";
import Menu from "@mui/material/Menu";
import ExitToAppIcon from "@mui/icons-material/ExitToApp";
import Logout from "../../Authorization/Logout";
import { UserQuery } from "../../../generated/graphql";

interface IProfile {
  data: UserQuery;
}

const style = {
  menuItem: { mb: "10px" },
  listItemText: { mr: "20px" },
  box: {
    width: "max-content",
    ml: "16px",
  },
  avatar: {
    width: 40,
    height: 40,
  },
  typography: {
    display: { xs: "none", md: "block" },
  },
};

const Profile: React.FC<IProfile> = (props) => {
  const { data } = props;

  const settings = [
    {
      title: "Настройки",
      icon: <ExitToAppIcon />,
    },
  ];

  const [anchorElUser, setAnchorElUser] =
    React.useState<null | HTMLElement>(null);

  const handleOpenProfile = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElUser(event.currentTarget);
  };

  const handleClickSettingsProfile = () => {
    setAnchorElUser(null);
  };

  return (
    <>
      <Tooltip title="Open settings">
        <Button onClick={handleOpenProfile}>
          <Avatar
            sx={style.avatar}
            alt="Remy Sharp"
            src="/static/images/avatar/1.jpg"
          />
          <Box sx={style.box}>
            <Typography sx={style.typography} variant="subtitle2">
              {data?.user?.firstName}
            </Typography>
            <Typography sx={style.typography} variant="subtitle2">
              {data?.user?.lastName}
            </Typography>
          </Box>
        </Button>
      </Tooltip>
      <Menu
        anchorEl={anchorElUser}
        keepMounted
        open={Boolean(anchorElUser)}
        onClose={handleClickSettingsProfile}
      >
        {settings.map((setting, index) => {
          const { icon, title } = setting;

          return (
            <MenuList key={index}>
              <MenuItem
                onClick={handleClickSettingsProfile}
                sx={style.menuItem}
              >
                <ListItemText sx={style.listItemText}>{title}</ListItemText>
                <ListItemIcon>{icon}</ListItemIcon>
              </MenuItem>
              <Logout setAnchorElUser={setAnchorElUser} />
            </MenuList>
          );
        })}
      </Menu>
    </>
  );
};

export default Profile;
