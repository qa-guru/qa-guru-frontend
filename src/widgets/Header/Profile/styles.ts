import { black } from "../../../theme/colors";

export const style = {
  menuItem: { mb: "10px" },
  listItemText: { mr: "20px" },
  box: {
    width: "max-content",
    ml: "16px",
  },
  typography: {
    display: { xs: "none", md: "block" },
    color: black.main,
  },
};
