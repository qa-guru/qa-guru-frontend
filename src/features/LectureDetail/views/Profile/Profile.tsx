import React from "react";
import { Box, Stack, Typography } from "@mui/material";
import { format, parseISO } from "date-fns";
import { IProfile } from "./Profile.types";
import AvatarCustom from "../../../../shared/AvatarCustom";

const Profile: React.FC<IProfile> = (props) => {
  const { firstName, lastName, date } = props;

  const fullName = `${firstName} ${lastName}`;

  return (
    <Stack spacing={1.7} direction="row" alignItems="center">
      <AvatarCustom fullName={fullName} />
      <Box>
        <Typography variant="subtitle1">{fullName}</Typography>
        <Stack direction="row" justifyContent="space-between">
          <Typography variant="subtitle2">
            {format(parseISO(date), "dd.MM.yyyy '|' HH:mm")}
          </Typography>
        </Stack>
      </Box>
    </Stack>
  );
};

export default Profile;
