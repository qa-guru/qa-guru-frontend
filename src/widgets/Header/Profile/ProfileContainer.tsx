import React from "react";
import Profile from "./Profile";
import { useUserQuery } from "../../../api/graphql/user/user";
import Spinner from "../../../shared/Spinner";
import NoDataErrorMessage from "../../../shared/NoDataErrorMessage";

const ProfileContainer: React.FC = () => {
  const { loading, data } = useUserQuery();

  if (loading) return <Spinner />;
  if (!data) return <NoDataErrorMessage />;

  return <Profile data={data} />;
};

export default ProfileContainer;
