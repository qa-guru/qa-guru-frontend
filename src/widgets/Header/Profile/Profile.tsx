import React from "react";
import {
  Box,
  Button,
  ListItemIcon,
  ListItemText,
  MenuItem,
  MenuList,
  Typography,
} from "@mui/material";
import Tooltip from "@mui/material/Tooltip";
import Menu from "@mui/material/Menu";
import SettingsIcon from "@mui/icons-material/Settings";
import { IProfile } from "./Profile.types";
import { style } from "./styles";
import Logout from "../../../features/Authorization/containers/LogoutContainer";
import AvatarCustom from "../../../shared/AvatarCustom";

const Profile: React.FC<IProfile> = (props) => {
  const { firstName, lastName } = props.data.user!;
  const fullName = `${firstName} ${lastName}`;

  const settings = [
    {
      title: "Настройки",
      icon: <SettingsIcon />,
    },
  ];

  const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(
    null
  );

  const handleOpenProfile = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElUser(event.currentTarget);
  };

  const handleClickSettingsProfile = () => {
    setAnchorElUser(null);
  };

  return (
    <>
      <Tooltip title="Open settings">
        <Button variant="text" onClick={handleOpenProfile}>
          <AvatarCustom fullName={fullName} />
          <Box sx={style.box}>
            <Typography sx={style.typography} variant="subtitle2">
              {firstName}
            </Typography>
            <Typography sx={style.typography} variant="subtitle2">
              {lastName}
            </Typography>
          </Box>
        </Button>
      </Tooltip>
      <Menu
        anchorEl={anchorElUser}
        keepMounted
        open={Boolean(anchorElUser)}
        onClose={handleClickSettingsProfile}
      >
        {settings.map((setting, index) => {
          const { icon, title } = setting;

          return (
            <MenuList key={index}>
              <MenuItem
                onClick={handleClickSettingsProfile}
                sx={style.menuItem}
              >
                <ListItemText sx={style.listItemText}>{title}</ListItemText>
                <ListItemIcon>{icon}</ListItemIcon>
              </MenuItem>
              <Logout setAnchorElUser={setAnchorElUser} />
            </MenuList>
          );
        })}
      </Menu>
    </>
  );
};

export default Profile;
