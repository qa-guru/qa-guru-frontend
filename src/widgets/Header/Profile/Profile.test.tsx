import { MemoryRouter } from "react-router-dom";
import Profile from "./Profile";
import { render } from "../../../test/utilsTest";
import { user } from "../../../shared/mocks/user.mock";

describe("Profile", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Profile data={user} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
