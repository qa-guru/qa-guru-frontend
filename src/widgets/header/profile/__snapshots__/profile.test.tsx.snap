// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile > renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="Open settings"
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1joc3qd-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element="true"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiBox-root css-1j85rlp"
    >
      <div
        class="MuiStack-root css-czl7p-MuiStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1d0j7ho-MuiAvatar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-4ozz3r-MuiTypography-root"
          >
            ss
          </h6>
        </div>
        <div
          class="MuiBox-root css-189yone"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-158lv7p-MuiTypography-root"
          >
            string string
          </p>
        </div>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
